data-processor/src/test/groovy/io/micronaut/data/processor/sql/JsonViewSpec.groovy

```diff
@@ -0,0 +1,84 @@
+
+    Long updateContactView(@Id Long id, String name);
+}
+""")
+
+        def findByIdQuery = getQuery(repository.getRequiredMethod("findById", Long))
+        def saveQuery = getQuery(repository.getRequiredMethod("save", ContactView))
+        def updateQuery = getQuery(repository.getRequiredMethod("update", ContactView))
+        def updateContactViewQuery = getQuery(repository.getRequiredMethod("updateContactView", Long, String))
+        def deleteByIdQuery = getQuery(repository.getRequiredMethod("deleteById", Long))
+        def deleteQuery = getQuery(repository.getRequiredMethod("delete", ContactView))
+        def deleteAllQuery = getQuery(repository.getRequiredMethod("deleteAll"))
+        def deleteAllIterableQuery = getQuery(repository.getRequiredMethod("deleteAll", Iterable<ContactView>))
+
+        expect:
+        findByIdQuery == 'SELECT cv.* FROM CONTACT_VIEW cv WHERE (cv.DATA.id = ?)'
+        saveQuery == 'INSERT INTO CONTACT_VIEW VALUES (?)'
+        updateQuery == 'UPDATE CONTACT_VIEW cv SET cv.DATA=? WHERE (cv.DATA.id = ?)'
+        updateContactViewQuery == 'UPDATE CONTACT_VIEW cv SET cv.name=? WHERE (cv.DATA.id = ?)'
+        deleteByIdQuery == 'DELETE  FROM CONTACT_VIEW  cv WHERE (cv.DATA.id = ?)'
+        deleteQuery == 'DELETE  FROM CONTACT_VIEW  cv WHERE (cv.DATA.id = ?)'
+        deleteAllQuery == 'DELETE  FROM CONTACT_VIEW  cv'
+        deleteAllIterableQuery == 'DELETE  FROM CONTACT_VIEW  cv WHERE (cv.DATA.id IN (?))'
+    }
+
+    void "test JsonView repository not supported property projection"() {
+        when:
+        buildRepository('test.ContactViewRepository', """
+import io.micronaut.data.jdbc.annotation.JdbcRepository;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+import io.micronaut.data.repository.GenericRepository;
+import io.micronaut.data.tck.entities.ContactView;
+@JdbcRepository(dialect = Dialect.ORACLE)
+interface ContactViewRepository extends GenericRepository<ContactView, Long> {
+
+    String findNameById(Long id);
+}
+""")
+
+        then:
+        def thrown = thrown(RuntimeException)
+        thrown.message.contains("Property name projection in entity ContactView not supported for JsonView and dialect ORACLE")
+    }
+
+    void "test JsonView repository not supported max function projection"() {
+        when:
+        buildRepository('test.ContactViewRepository', """
+import io.micronaut.data.jdbc.annotation.JdbcRepository;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+import io.micronaut.data.repository.GenericRepository;
+import io.micronaut.data.tck.entities.ContactView;
+@JdbcRepository(dialect = Dialect.ORACLE)
+interface ContactViewRepository extends GenericRepository<ContactView, Long> {
+
+    int findMaxAgeByName(String name);
+}
+""")
+
+        then:
+        def thrown = thrown(RuntimeException)
+        thrown.message.contains("Function MAX projection for property age in entity ContactView not supported for JsonView and dialect ORACLE")
```

Comment:
According to this https://docs.oracle.com/en/database/oracle/oracle-database/21/adjsn/json-path-expressions.html#GUID-8656CAB9-C293-4A99-BB62-F38F3CFC4C13 it should work

Reply:
I have tried that. For query 
```
SELECT MAX(student_view_.DATA.averageGrade) FROM STUDENT_VIEW student_view_
```
when reading field, Oracle resultset throws this exception:

```
java.sql.SQLException: ORA-17004: Invalid column type: getDouble not implemented for class oracle.jdbc.driver.T4CJsonAccessor
	at app//oracle.jdbc.driver.GeneratedAccessor.getDouble(GeneratedAccessor.java:339)
	at app//oracle.jdbc.driver.GeneratedStatement.getDouble(GeneratedStatement.java:160)
	at app//oracle.jdbc.driver.GeneratedScrollableResultSet.getDouble(GeneratedScrollableResultSet.java:226)
	at app//io.micronaut.data.jdbc.mapper.ColumnIndexResultSetReader.readDouble(ColumnIndexResultSetReader.java:182)
```

Reply:
Can you try `student_view_.DATA.averageGrade.maxNumber()` or `.max()`?

Reply:
That works but returns averageGrade for all items found in the Json View. Seems like it is not meant to be used for projections and max/min/avg etc. functions?

Reply:
We can add aggregation later

Reply:
It is solved by using `.numberOnly()` and `.stringOnly()` both for projection and aggregates. Just need to add support for boolean and see about date/time if there is support.